src/shell/boards/board-container.tsx

```diff
@@ -155,62 +201,172 @@ export const BoardContainer: FC = () => {
 		[boards, current, orderedBoards]
 	);
 
-	if (isEmpty(boards) || !current) return null;
+	const boardRef = useRef<HTMLDivElement>(null);
+	const boardContainerRef = useRef<HTMLDivElement>(null);
+	const [lastSavedBoardSize, setLastSavedBoardSize] = useLocalStorage<Partial<SizeAndPosition>>(
+		LOCAL_STORAGE_BOARD_SIZE,
+		{}
+	);
+	const [currentBoardSize, setCurrentBoardSize] = useState(lastSavedBoardSize);
+	const lastSavedBoardSizeRef = useRef(lastSavedBoardSize);
+
+	useEffect(() => {
+		setCurrentBoardSize(lastSavedBoardSize);
+		lastSavedBoardSizeRef.current = lastSavedBoardSize;
+	}, [lastSavedBoardSize]);
+
+	const isDefaultSizeAndPos = useMemo(() => size(currentBoardSize) === 0, [currentBoardSize]);
+
+	const resetSizeAndPosition = useCallback(() => {
+		setLastSavedBoardSize({});
+		reduceBoards();
+	}, [setLastSavedBoardSize]);
+
+	useEffect(() => {
+		// reset position when the board is closed
+		if (isEmpty(boards)) {
```

Comment:
Maybe we can memoize the isEmpty(boards) info and use it as dependency to improve understanding of effect trigger